analyzers/src/SonarAnalyzer.Common/Helpers/MethodParameterLookupBase.cs

```diff
@@ -18,115 +18,139 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-namespace SonarAnalyzer.Helpers
+using System.Linq;
+
+namespace SonarAnalyzer.Helpers;
+
+public interface IMethodParameterLookup
+{
+    bool TryGetSymbol(SyntaxNode argument, out IParameterSymbol parameter);
+    bool TryGetSyntax(IParameterSymbol parameter, out ImmutableArray<SyntaxNode> expressions);
+    bool TryGetSyntax(string parameterName, out ImmutableArray<SyntaxNode> expressions);
+    bool TryGetNonParamsSyntax(IParameterSymbol parameter, out SyntaxNode expression);
+}
+
+// This should come from the Roslyn API (https://github.com/dotnet/roslyn/issues/9)
+internal abstract class MethodParameterLookupBase<TArgumentSyntax> : IMethodParameterLookup
+    where TArgumentSyntax : SyntaxNode
 {
-    public interface IMethodParameterLookup
+    private readonly SeparatedSyntaxList<TArgumentSyntax>? argumentList;
+
+    protected abstract SyntaxToken? GetNameColonArgumentIdentifier(TArgumentSyntax argument);
+    protected abstract SyntaxNode Expression(TArgumentSyntax argument);
+
+    public IMethodSymbol MethodSymbol { get; }
+    private SymbolInfo? MethodSymbolInfo { get; }
+
+    protected MethodParameterLookupBase(SeparatedSyntaxList<TArgumentSyntax>? argumentList, SymbolInfo? methodSymbolInfo)
+        : this(argumentList, methodSymbolInfo?.Symbol as IMethodSymbol)
```

Comment:
This looks wrong to me. Shouldn't the MethodParameterLookupBase be extended to handle a SymbolInfo instead (or a method group `IEnumerable<IMethodSymbol>`)? Let's discuss this tomorrow online.